ngraph_creator/operations/src/Split.cpp

```diff
@@ -0,0 +1,77 @@
+#include <Split.hpp>
+
+namespace android {
+namespace hardware {
+namespace neuralnetworks {
+namespace nnhal {
+
+Split::Split(int operationIndex) : OperationsBase(operationIndex) {
+    mDefaultOutputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, 0);
+}
+
+bool Split::validate() {
+    // check output type
+    if (!checkOutputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32) &&
+        !checkOutputOperandType(0, (int32_t)OperandType::TENSOR_INT32)) {
+        return false;
+    }
+
+    // Check all input types
+    if (!checkInputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32) &&
+        !checkInputOperandType(0, (int32_t)OperandType::TENSOR_INT32)) {
+        return false;
+    }
+
+    if (!checkInputOperandType(1, (int32_t)OperandType::INT32)) {
+        return false;
+    }
+
+    if (!checkInputOperandType(2, (int32_t)OperandType::INT32)) {
+        return false;
+    }
+
+    return true;
+}
+
+void Split::connectOperationToGraph() { createNode(); }
+
+std::shared_ptr<ngraph::Node> Split::createNode() {
+    // Creating input nodes
+    auto operandType = sModelInfo->getOperandType(0);
+
+    std::shared_ptr<ngraph::Node> splitNode;
+
+    if (operandType == OperandType::TENSOR_FLOAT32) {
+        splitNode = getInputNode<float>(0);
+    }
+
+    if (operandType == OperandType::TENSOR_INT32) {
+        splitNode = getInputNode<int>(0);
+    }
+
+    auto axis = sModelInfo->ParseOperationInput<int>(mNnapiOperationIndex, 1);
+    auto axisNode = ngraph::opset3::Constant::create(ngraph::element::i64, {}, {axis});
+    auto numSplits = sModelInfo->ParseOperationInput<uint32_t>(mNnapiOperationIndex, 2);
+
+    auto outputNode =
+        std::make_shared<ngraph::opset3::Split>(splitNode, axisNode, numSplits)->outputs();
+
+    for (int i = 0; i < numSplits; i++) {
+        auto outputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, i);
+        // TODO: remove this dummy convert
+        auto outNode =
+            std::make_shared<ngraph::opset3::Convert>(outputNode[i], ngraph::element::f32);
```

Comment:
Instead of this, can you try this :
Inside the below Output check, create a Result Node from outNode, and assign that to outputIndex.

`auto resultNode = std::make_shared<ngraph::opset3::Result>(outNode);`
`addResultNode(outputIndex, resultNode);`
`mNgraphNodes->setOutputAtOperandIndex(outputIndex, resultNode);`

Reply:
Looks like this is causing some issue with fetching the corresponding blob.